src/main/scala/mesosphere/marathon/core/leadership/impl/AbdicateOnConnectionLossActor.scala

```diff
@@ -0,0 +1,40 @@
+package mesosphere.marathon.core.leadership.impl
+
+import akka.actor.{ Actor, ActorLogging, Props }
+import com.twitter.common.zookeeper.ZooKeeperClient
+import mesosphere.marathon.LeadershipAbdication
+import org.apache.zookeeper.{ WatchedEvent, Watcher }
+
+object AbdicateOnConnectionLossActor {
+  def props(zk: ZooKeeperClient, leader: LeadershipAbdication): Props = {
+    Props(new AbdicateOnConnectionLossActor(zk, leader))
+  }
+}
+
+class AbdicateOnConnectionLossActor(zk: ZooKeeperClient,
+                                    leader: LeadershipAbdication) extends Actor with ActorLogging {
+
+  private[impl] val watcher = new Watcher {
+    val reference = self
+    override def process(event: WatchedEvent): Unit = reference ! event
+  }
+
+  override def preStart(): Unit = {
+    log.info("Register as ZK Listener")
+    zk.register(watcher)
```

Comment:
To avoid race-conditions, you should test the current zookeeper connection state after registering for events. The connection might already have died.